ddtrace/tracer.py

```diff
@@ -351,13 +351,21 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
             trace_id = context.trace_id
             parent_span_id = context.span_id
 
-        if trace_id:
-            # child_of a non-empty context, so either a local child span or from a remote context
-
-            # when not provided, inherit from parent's service
+        # The following precedence is used for a new span's ``service``:
```

Comment:
we might want to consider reformatting this comment, the information is correct, but it doesn't really line up with the code very well.

e.g.

```python
# The following .... : 
# 1. Explicitly provide service name
#     a. User provided or integration provided service name
# 2. Parent's service name (if exists)
# 3. Globally configured service name
#     a. ``config.service``/`DD_SERVICE`
```

Reply:
I like yours much better 😄.